catalyst/dl/experiment/gan.py

```diff
@@ -0,0 +1,158 @@
+from collections import OrderedDict
+from typing import Dict, List, Union
+
+from torch.utils.data import DataLoader
+
+from catalyst.dl import Callback
+from catalyst.dl.callbacks import (
+    CheckpointCallback, ConsoleLogger, PhaseBatchWrapperCallback,
+    PhaseManagerCallback, RaiseExceptionCallback, VerboseLogger
+)
+from catalyst.utils.tools.typing import Criterion, Model, Optimizer, Scheduler
+from .base import BaseExperiment
+
+
+class GanExperiment(BaseExperiment):
+    """
+    One-staged GAN experiment
+    Mainly used in notebooks
+    """
+
+    def __init__(
+        self,
+        model: Union[Model, Dict[str, Model]],
+        loaders: "OrderedDict[str, DataLoader]",
+        callbacks: "OrderedDict[str, Callback]" = None,
```

Comment:
we also should support a `List[Callback]`, like in `SupervisedRunner.train`

Reply:
Ok, but the main reason of GanExperiment is about callback wrapping, and it's strange if you wrap something without knowing how to address it(key). Added it to type hint anyway.